bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/handler/IpCameraHandler.java

```diff
@@ -0,0 +1,2020 @@
+    public Double motionThreshold = 0.0016;
+    public int audioThreshold = 35;
+    @SuppressWarnings("unused")
+    private @Nullable StreamServerHandler streamServerHandler;
+    boolean streamingSnapshotMjpeg = false;
+    public boolean motionAlarmEnabled = false;
+    public boolean audioAlarmEnabled = false;
+    public boolean ffmpegSnapshotGeneration = false;
+    public boolean snapshotPolling = false;
+    public OnvifConnection onvifCamera = new OnvifConnection(this, "", "", "");
+
+    public IpCameraHandler(Thing thing) {
+        super(thing);
+        config = thing.getConfiguration();
+    }
+
+    private IpCameraHandler getHandle() {
+        return this;
+    }
+
+    // false clears the stored user/pass hash, true creates the hash
+    public boolean setBasicAuth(boolean useBasic) {
+        if (useBasic == false) {
+            logger.debug("Clearing out the stored BASIC auth now.");
+            basicAuth = "";
+            return false;
+        } else if (!basicAuth.equals("")) {
+            // due to camera may have been sent multiple requests before the auth was set, this may trigger falsely.
+            logger.warn("Camera is reporting your username and/or password is wrong.");
+            return false;
+        }
+        if (!username.equals("") && !password.equals("")) {
+            String authString = username + ":" + password;
+            ByteBuf byteBuf = null;
+            try {
+                byteBuf = Base64.encode(Unpooled.wrappedBuffer(authString.getBytes(CharsetUtil.UTF_8)));
+                basicAuth = byteBuf.getCharSequence(0, byteBuf.capacity(), CharsetUtil.UTF_8).toString();
+            } finally {
+                if (byteBuf != null) {
+                    byteBuf.release();
+                    byteBuf = null;
+                }
+            }
+            return true;
+        } else {
+            cameraConfigError(
+                    "Camera is asking for Basic Auth when you have not provided a username and/or password !");
+        }
+        return false;
+    }
+
+    private String getCorrectUrlFormat(String longUrl) {
+        String temp = longUrl;
+        URL url;
+
+        try {
+            url = new URL(longUrl);
+            int port = url.getPort();
+            if (port == -1) {
+                if (url.getQuery() == null) {
+                    temp = url.getPath();
+                } else {
+                    temp = url.getPath() + "?" + url.getQuery();
+                }
+            } else {
+                if (url.getQuery() == null) {
+                    temp = ":" + url.getPort() + url.getPath();
+                } else {
+                    temp = ":" + url.getPort() + url.getPath() + "?" + url.getQuery();
+                }
+            }
+        } catch (MalformedURLException e) {
+            if (!longUrl.equals("ffmpeg")) {
+                logger.error("A non valid url has been given to the binding, it is: {}", longUrl);
```

Comment:
Fixed by removing error and changing thing status instead. In this case it would happen if someone entered a typo in the URL. Instead of 'http://hostname/snap.jpg' they entered 'HTPP://hostname/snap.jpg'.

Reply:
https://github.com/openhab/openhab-addons/pull/8329/commits/b1c023a60b67f2cf87243994b89e83a6256cf260